backend/services/notificationService.ts

```diff
@@ -1,60 +1,102 @@
       personalisation,
       reference: null,
     })
   }
 
-  private async getUserDetails(context: Context, username: string) {
+  private async getUserDetails(context: Context, username: string): Promise<UserDetails> {
     const [{ email }, { name }] = await Promise.all([
       this.oauthApi.userEmail(context, username),
       this.oauthApi.userDetails(context, username),
     ])
     return { email, name }
   }
 
-  private getEmailAddress(recipient: Recipient, agencyId: string, userEmail: string): string {
+  private async getEmailAddress(
+    context: Context,
+    recipient: Recipient,
+    agencyId: string,
+    userEmail: string
+  ): Promise<string> {
     switch (recipient) {
       case 'user':
         return userEmail
       case 'omu':
-        return notifications.emails[agencyId]?.omu
+        return this.prisonRegisterApi.getOffenderManagementUnitEmailAddress(context, agencyId)
       case 'vlb':
-        return notifications.emails[agencyId]?.vlb
+        return this.prisonRegisterApi.getVideoLinkConferencingCentreEmailAddress(context, agencyId)
       default:
         throw new Error(`could not send email to ${recipient}`)
     }
   }
 
-  private async sendEmails(context: Context, username: string, spec: EmailSpec): Promise<void> {
-    const { email: userEmail, name: usersName } = await this.getUserDetails(context, username)
+  private async sendEmails(context: Context, username: string, emailSpec: EmailSpec): Promise<void> {
+    const userDetails = await this.getUserDetails(context, username)
+    const recipientEmailer = recipientEmailSpec =>
+      this.emailToRecipient(context, recipientEmailSpec, userDetails, emailSpec)
+    await Promise.all(emailSpec.recipients.map(recipientEmailer))
+  }
+
+  private async emailToRecipient(
+    context: Context,
+    recipientEmailSpec: RecipientEmailSpec,
+    userDetails: UserDetails,
+    emailSpec: EmailSpec
+  ): Promise<void> {
+    try {
+      const emailAddress = await this.getEmailAddress(
+        context,
+        recipientEmailSpec.recipient,
+        emailSpec.agencyId,
+        userDetails.email
+      )
 
-    spec.recipients.forEach(email => {
-      const emailAddress = this.getEmailAddress(email.recipient, spec.agencyId, userEmail)
-      if (emailAddress) {
-        this.sendEmail({
-          templateId: email.template,
-          email: emailAddress,
-          personalisation: email.personalisation(usersName),
-        }).catch(error => {
-          log.error(
-            `Failed to email the ${email.recipient} a ${spec.name}: ${error.message}`,
-            error.response?.data?.errors
-          )
-        })
+      if (!emailAddress) {
+        return Promise.resolve()
       }
-    })
+
+      return this.sendEmail({
+        templateId: recipientEmailSpec.template,
+        emailAddress,
+        personalisation: recipientEmailSpec.personalisation(userDetails.name),
+      })
+    } catch (error) {
+      log.error(
```

Comment:
This log statement was present before DCS-1055.  No new logging here.
The only new logging takes place within prisonRegisterApi. This logging is actually done by oauthEnabledClient.ts, the client used by prisonRegisterApi.  Removing this logging would require changing oauthEnabledClient.ts or creating a separate ouath enabled client specifically for this case